dataprofiler/tests/profilers/test_structured_options.py

```diff
@@ -0,0 +1,153 @@
+    
+        # Treat is_enabled as a BooleanOption
+        for key in self.keys:
+            expected_error = "type object '{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option.set({'{}.is_enabled.is_enabled'.format(key): True})
+    
+    def test_validate_helper(self):
+        # Valid cases should return [] while invalid cases should return a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+
+        # Default Configuration Is Valid
+        self.assertEqual([], option._validate_helper())
+        
+        # Variable Path Is Not A String
+        expected_error = "The variable path must be a string."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            option._validate_helper(1)
+        
+        # Option is_enabled is not a boolean
+        for key in self.keys:
+            option.set({'{}.is_enabled'.format(key): "Hello World"}) 
+        expected_error = ['{}.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in self.keys]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option._validate_helper())))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.int = StructuredOptions()
+        option.float = StructuredOptions()
+        option.datetime = StructuredOptions()
+        option.text = StructuredOptions()
+        option.order = StructuredOptions()
+        option.category = StructuredOptions()
+        option.data_labeler = StructuredOptions()
+
+        expected_error = set()
+        for key in self.keys:
+            ckey = key.capitalize()
+            if key == "data_labeler": ckey = "DataLabeler"
+            elif key == "category": ckey = "Categorical"
+            elif key == "datetime": ckey = "DateTime"
+            expected_error.add('{}.{} must be a(n) {}Options.'.format(optpth, key, ckey))
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, set(option._validate_helper()))
+            
+    def test_validate(self):
+        # Valid cases should return None while invalid cases should return or throw a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+    
+        # Default Configuration Is Valid
+        self.assertEqual(None, option.validate())
+        
+        # Option is_enabled is not a boolean
+        for key in self.keys:
+            option.set({'{}.is_enabled'.format(key): "Hello World"}) 
+        
+        expected_error = ["{}.{}.is_enabled must be a Boolean.".format(optpth, key) 
+            for key in self.keys]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option.validate(raise_error=False))))
```

Comment:
What is the point of asserting that set A is equal to set A intersect set B? Why not just ensure that set A equals set B?

Reply:
This pattern is present several times, I think all should be fixed unless I'm missing something.

Reply:
Setting is_enabled doesn't just change is_enabled for each of those options but also the properties of those options as well. This means that the cardinality of set B >= set A, but we are only concerned with whether set A is a subset of set B

Reply:
Ah I see, that makes sense, and I see there is no .assertIsSubset or anything, so I suppose this is fine, I would maybe just appreciate a comment above to explain what exactly you are trying to verify.

Reply:
Sure, I will add a comment to clarify in the next push